cxx-qt-lib/build.rs

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use quote::ToTokens;
+use std::io::Write;
+
+fn main() {
+    let opt = cxx_gen::Opt::default();
+
+    // Read the cargo folder and out folder
+    let dir_manifest = std::env::var("CARGO_MANIFEST_DIR").expect("Could not get manifest dir");
+    let dir_out = std::env::var("OUT_DIR").expect("Could not get out dir");
+
+    // Prepare directories we'll write to
+    let include_path = format!("{}/cxx-qt-lib/include", dir_out);
+    std::fs::create_dir_all(&include_path).expect("Could not create cxx-qt-lib include dir");
+    let src_path = format!("{}/cxx-qt-lib/src", dir_out);
+    std::fs::create_dir_all(&src_path).expect("Could not create cxx-qt-lib sc dir");
+
+    let qfiles = vec!["qcolor", "qpoint", "qstring"];
+
+    for qfile in qfiles {
+        // Read the rust source files
+        let path = format!("{}/src/{}.rs", dir_manifest, qfile);
+        let content = std::fs::read_to_string(path).expect("Could not read Rust file");
+        let file = syn::parse_file(&content).unwrap();
+
+        // Generate the CXX header and cc for the file
+        let gen_result = cxx_gen::generate_header_and_cc(file.into_token_stream(), &opt)
+            .expect("Could not generate C++ from Rust file");
+
+        // Write out the CXX files
+        let h_path = format!("{}/{}_cxx.h", include_path, qfile);
+        let mut header = std::fs::File::create(&h_path).expect("Could not create header file");
+        header
+            .write_all(&gen_result.header)
+            .expect("Could not write header file");
+
+        let cpp_path = format!("{}/{}_cxx.cpp", src_path, qfile);
+        let mut cpp = std::fs::File::create(&cpp_path).expect("Could not create cpp file");
+        cpp.write_all(&gen_result.implementation)
+            .expect("Could not write cpp file");
+    }
+
+    // Copy the static source files into the directory
+    std::fs::copy(
+        format!("{}/{}", dir_manifest, "include/qt_types.h"),
+        format!("{}/qt_types.h", include_path),
+    )
+    .expect("Could not copy qt_types.h");
+    std::fs::copy(
+        format!("{}/{}", dir_manifest, "src/qt_types.cpp"),
+        format!("{}/qt_types.cpp", src_path),
+    )
+    .expect("Could not copy qt_types.cpp");
+
+    // Expose target directory to downstream crates
+    // https://github.com/rust-lang/cargo/issues/3544
+    println!("cargo:target_dir={}", dir_out);
```

Comment:
is there any other way of doing this ? this is the only weird part of this change :-)

Reply:
Instead we now use include_str and then expose this as a const to cxx-qt-build :-)